src/frontend/apps/desk/src/features/teams/components/Member/ModalRole.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+  Button,
+  Modal,
+  ModalSize,
+  Radio,
+  RadioGroup,
+  VariantType,
+  useToastProvider,
+} from '@openfun/cunningham-react';
+import { t } from 'i18next';
+import { useState } from 'react';
+
+import { Box, Text, TextErrors } from '@/components';
+import { useAuthStore } from '@/features/auth';
+import { Access, Role, useUpdateTeamAccess } from '@/features/teams/api/';
+
+interface ModalRoleProps {
+  access: Access;
+  onClose: () => void;
+  teamId: string;
+}
+
+export const ModalRole = ({ onClose, teamId, access }: ModalRoleProps) => {
+  const [localRole, setLocalRole] = useState(access.role);
+  const { userData } = useAuthStore();
+  const { toast } = useToastProvider();
+  const {
+    mutate: updateTeamAccess,
+    error: errorUpdate,
+    isError: isErrorUpdate,
+  } = useUpdateTeamAccess({
+    onSuccess: () => {
+      toast(t('The role has been updated'), VariantType.SUCCESS, {
+        duration: 4000,
+      });
+      onClose();
+    },
+  });
+
+  const isOtherOwner =
+    access.role === Role.OWNER && userData?.id !== access.user.id;
+  const isNotAllowed = isOtherOwner;
+
+  return (
+    <Modal
+      isOpen
+      leftActions={
+        <Button color="secondary" fullWidth onClick={() => onClose()}>
+          {t('Cancel')}
+        </Button>
+      }
+      onClose={() => onClose()}
+      rightActions={
+        <Button
+          color="primary"
+          fullWidth
+          onClick={() => {
+            updateTeamAccess({
+              role: localRole,
+              teamId,
+              accessId: access.id,
+            });
+          }}
+          disabled={isNotAllowed}
+        >
+          {t('Validate')}
+        </Button>
+      }
+      size={ModalSize.MEDIUM}
+      title={t('Update the role')}
+    >
+      <Box aria-label={t('Radio buttons to update the roles')}>
+        {isErrorUpdate && (
+          <TextErrors className="mb-s" causes={errorUpdate.cause} />
+        )}
+
+        {isOtherOwner && (
+          <Text
+            $theme="warning"
+            $direction="row"
+            $align="center"
+            $gap="0.5rem"
+            className="mb-t"
+            $justify="center"
+          >
+            <span className="material-icons">warning</span>
+            {t('You cannot update the role of other owner.')}
+          </Text>
+        )}
+        <RadioGroup>
```

Comment:
Based on `abilities['set_role_to']` values, you can guess which radio item is allowed or not.

Reply:
https://github.com/numerique-gouv/people/pull/73/commits/8267fe3683a10b980d7df4aaf5b02b25f701054d